lib/runner/extensions/request.command.js

```diff
@@ -1,4 +1,42 @@
-var _ = require('lodash');
+var _ = require('lodash'),
+    sdk = require('postman-collection'),
+
+    createItemContext = require('../create-item-context').createItemContext,
+
+    /**
+     * Resolve variables in item and auth in context.
+     *
+     * @param {Object} context
+     * @param {Item} [context.item]
+     * @param {RequestAuth} [context.auth]
+     * @param {Object} payload
+     * @param {VariableScope} payload._variables
+     * @param {VariableScope} payload.data
+     * @param {VariableScope} payload.environment
+     * @param {VariableScope} payload.collectionVariables
+     * @param {VariableScope} payload.globals
+     */
+    resolveVariables = function (context, payload) {
+        if (!context.item) { return; }
+
+        // @todo - resolve variables in a more graceful way
+
+        // @todo - no need to sync variables when SDK starts supporting resolution from scope directly
+        context.item = new sdk.Item(context.item.toObjectResolved(null, [payload._variables.syncVariablesTo({}),
+            payload.data, payload.environment.syncVariablesTo({}), payload.collectionVariables.syncVariablesTo({}),
+            payload.globals.syncVariablesTo({})], {ignoreOwnVariables: true}));
+
+        var item = context.item,
+            auth = context.auth;
+
+        // Re-parse the URL, because variables have been resolved now, and things might be moved around
+        item.request.url = new (sdk.Url)(item.request.url.toString());
+
+        // resolve variables in auth
+        auth && (context.auth = new sdk.RequestAuth(auth.toObjectResolved(null, [payload._variables.syncVariablesTo({}),
```

Comment:
This and the section above could be extracted into a common method.

Reply:
We'll be adding a proper controller for resolving variables. Otherwise, it wouldn't give us much of organization.